src/metabase/search/scoring.clj

```diff
@@ -171,12 +178,12 @@
 (defn- combined-score
   [{:keys [text-score result]}]
   [(pinned-score result)
-   (- text-score)
-   (model->sort-position (:model result))
-   (:name result)])
+   (dashboard-count-score result)
+   text-score
+   (model->sort-position (:model result))])
```

Comment:
`text-score` can stay positive since now things are sorted in the opposite order.

I removed the `:name` since it'd end up putting things in z–a order and I highly doubt it's a meaningful tie-breaker anyway. It was just a carryover from before we had the other scores.

Reply:
I'm confused by your sorting. I think the priorityqueue is the only thing doing the sorting by using the compare function you passed in. it should return its results sorted from lowest to highest. I'm not following why you're ordering things and putting negative numbers in places.

Reply:
I think thinking in terms of a sort order is misleading. Just think in terms of a compare that says which of two things is bigger. Don't think of order, sequence, or whatever. The priority queue just keeps the biggest N items. It also returns them sorted from lowest to highest.

```clojure
  ([^PriorityQueue q]
   (loop [acc []]
     (if-let [x (.poll q)]
       (recur (conj acc x))
       acc)))
```
That's how the priority queue is drained. Poll will pull the least element, and that is accumulated so you get `[low ... high]` returned. You can just reverse this if you want sorted from highest to lowest. So order is guaranteed. I'm confused why you want a reversed comparator for sorting outside of the PQ and a non-reversed one inside the PQ. Just a single function that says which is bigger allows the PQ to accumulate only the biggest N items and then return them sorted from lowest to highest. Reverse if you like.